main/2021/CG-08-03.md

```diff
@@ -30,12 +30,12 @@ Installation is required, see the calendar invite.
     1. Summary of [interface-types/#135](https://github.com/WebAssembly/interface-types/issues/135) and discussion [40 min].
         1. Poll for supporting [UTF-16](https://github.com/WebAssembly/interface-types/issues/136#issuecomment-861799460) lifting and lowering
         1. Poll for supporting compact UTF-16 (Latin1) lifting and lowering
-        1. Discussion: Is the Component Model in-line with our high-level goals?
-           * i.e. "integrate well with the existing Web platform"
-           * i.e. "execute in the same semantic universe as JavaScript"
-           * i.e. "maintain the backwards-compatible nature of the Web"
-           * i.e. "promote other compilers and tools targeting WebAssembly"
-           * i.e. "WebAssembly **also** supports non-web embeddings" (but not primarily)
+        1. Discussion: Is the Component Model in-line with our [high-level goals](https://webassembly.org/docs/high-level-goals/), with [what](https://webassembly.org/) we [communicate](https://www.w3.org/2020/03/webassembly-wg-charter), and [envisioned](https://www.youtube.com/watch?v=ZKydJPRosa0)? In particular:
```

Comment:
I'm not sure how productive this agenda item can be, given that we previously achieved consensus on pursuing the component model, but I don't have any further editorial suggestions.

Reply:
I would typically agree, but this was and still is a very controversial topic for a reason, that I think hasn't gotten our full attention because we are too afraid to address the elephant in the room. I think it can be healing for the group to finally speak this out, so we can eventually move on.

Reply:
I'd also like to use the opportunity here to respectfully question the consensus we achieved on the component model:

* In the next steps we voted on, it was claimed that "adapter functions [...] are just an optimization over using a fixed, canonical ABI", and I gave a reasonable, as I think, presentation why this is not the case for strings once the `char` type has been restricted to only allow specific semantics.
* Right before voting, the champion clarified upon request that the vote excluded string semantics, yet it turned out that it in fact directly implied string semantics as a result of the scope change. It has been argued that this was necessary given the new requirements, yet doing so implies that choosing DOMString-compatible semantics is off the table anyhow, so the clarification would not have been necessary in the first place, and it could just have been stated right away.

As such I think that the most productive we can do is to have more discussion, and perhaps undo a potentially problematic scope change if it turns out that the group has an interest to do so.

Reply:
The semantics of strings is a minor detail compared to the full scope of the component model we voted and achieved consensus on. I think it would set a bad precedent to allow that vote to be reconsidered. We should respect the decision of the community and continue working assuming the component model will have the scope and goals we voted on, no matter what our different technical opinions about string semantics and encodings.

Reply:
It seems that we disagree, and according to the process this is fine. But what do we do now? Propose a vote on whether to discuss the topic? What do we do when we disagree on adding it to the agenda as well? Propose a vote on whether to propose a vote on whether to discuss the topic? Have we maybe just found a reference cycle in the process? How do we invoke the backup collector? :)

Reply:
What about:

> Is a "single list-of-USVs string type" in-line with our [high-level goals](https://webassembly.org/docs/high-level-goals/), with [what](https://webassembly.org/) we [communicate](https://www.w3.org/2020/03/webassembly-wg-charter), and [envisioned](https://www.youtube.com/watch?v=ZKydJPRosa0)?

Reply:
Yes, this seems reasonable to me as part of the discussion of whether `string` should have list-of-USV semantics.

Reply:
I don't think the Discussion bullet accurately describes the discussion we need to have.  Given recent strong CG consensus on the goals and requirements of the component model and the fact that Interface Types and `string` are within the scope of the component model, the discussion is whether a single list-of-USV `string` is in line with the component model's goals and requirements.  Otherwise, we're just revisiting the component model's goals which, as others pointed out above, is not going to be productive.

Reply:
Can you explain your clarification before the CM vote, that we'd not vote on string semantics, then clarified?

Reply:
The vote was about the goals and requirements of the new component model (and whether to have one at all); things like cross-language composability, portability, virtualization, etc.  These set an overall context for making design choices like string semantics (and 100s of other design choices) which are, of course, highly context dependent.  Thus, the vote was on the context, not all the 100s of consequent design choices, and the clarification was meant to be clear on this, since it was an explicit concern you raised in an issue.

Reply:
I disagree that what you are saying is in-line with voting on a "general direction" yet not achieving consensus on the concrete "next steps". This is not how I understood it and if it would have been clear I would not have voted NEUTRAL to do you a favor. In fact, I was pressured, against my reading of what is at stake here, by several WG members not to block consensus on the general vote, which I would question separately if this is the outcome.

Reply:
I'm having a hard time parsing what precisely you are disagreeing with.  My claim is simply: there is strong CG consensus around the goals of the component model including cross-language interoperability and virtualizability, and thus these should be the starting points for any discussion about string semantics within the component model; we shouldn't start over from first principles.

Reply:
> In fact, I was pressured, against my reading of what is at stake here, by several WG members not to block consensus on the general vote, which I would question separately if this is the outcome.

@dcodeIO I take this to be referring to the many hours I spent in calls with you, giving advice on how to navigate this situation _to your advantage_, which you were very grateful for at the time. I don't appreciate your misrepresentation here.

I stand by my advice that it would have been poor behaviour to attempt to block the general component model vote on the grounds that the draft canonical ABI supported only UTF-8. To facilitate you, I [explicitly brought up](https://github.com/WebAssembly/meetings/blob/main/main/2021/CG-05-25.md#follow-up-discussion-from-the-cg-04-27-presentation-and-design1415-20-min) during the CG vote that we would need a separate future vote on the contents of the canonical ABI.

Reply:
I still appreciate all you have done, Conrad, and I respect you for your efforts. Wasm needs more people like you, but it also direly needs some honesty and integrity. In fact I didn't block the general component model vote in good faith in the hope that it would motivate constructive discussion on my concerns, but now look at the outcome: A follow-up meeting in a smaller group was suggested, but then was refused due to "reluctance". I've presented my concerns reasonably I think, yet was met with the same old fallacies. I've made concrete suggestions on how to improve the proposal, yet was met with omission. I suggested concrete follow-up action items, yet was gamed again. Needless to say that my earlier general concerns on Wasm's non-web language-exclusive trajectory remain largely unaddressed until today, also because an entire group of people decided to ghost me. And what's happening in this PR just makes me lose the remaining bits of trust into the process that you preserved in me, as well as in Wasm's goals, its vision, its principles, its values, and in particular its leadership.